File: restring/src/main/java/com/ice/restring/CustomResourcesContextWrapper.java
package com.ice.restring;

import android.content.Context;
import android.content.ContextWrapper;
import android.content.res.Resources;

/**
 * A context wrapper which provide another Resources instead of the original one.
 */
class CustomResourcesContextWrapper extends ContextWrapper {

    private Resources resources;

    public CustomResourcesContextWrapper(Context base, Resources resources) {
        super(base);
        this.resources = resources;
    }

    @Override
    public Resources getResources() {
        return resources;
    }
}

File: restring/src/main/java/com/b3nedikt/restring/CustomResourcesContextWrapper.kt
package com.b3nedikt.restring

import android.content.Context
import android.content.ContextWrapper
import android.content.res.Resources

/**
 * A context wrapper which provide another Resources instead of the original one.
 */
internal class CustomResourcesContextWrapper(
        base: Context,
        private val resources: Resources
) : ContextWrapper(base) {

    override fun getResources() = resources
}
